app/components/NoticeTypeCheckboxes.tsx

```diff
@@ -157,29 +159,161 @@ const NoticeTypes = {
   ],
 }
 
-interface NoticeTypeCheckboxeProps {
+type DataRow = {
+  topic: string
+  latestRate: number
+}
+
+type FormattedData = {
+  text: DataRow[]
+  binary: DataRow[]
+  voevent: DataRow[]
+}
+
+interface NoticeTypeCheckboxProps {
   defaultSelected?: string[]
+  selectedFormat?: 'binary' | 'text' | 'voevent'
+  showTotalSelected?: boolean
+}
+
+export async function loadTriggerRateData() {
+  return await getAverageTriggerRate()
 }
 
 export function NoticeTypeCheckboxes({
   defaultSelected,
-}: NoticeTypeCheckboxeProps) {
+  selectedFormat = 'text',
+  showTotalSelected = false,
+}: NoticeTypeCheckboxProps) {
+  const [metrics, setMetrics] = useState<FormattedData>()
+  const [userSelected, setUserSelected] = useState<{ [key: string]: boolean }>(
+    {}
+  )
+  const [selectedCounter, setSelectedCounter] = useState(0)
+  const [alertEstimate, setAlertEstimate] = useState(0)
```

Comment:
Initial state may be incorrect if any of the checkboxes are initially checked.

Reply:
I recently worked through this, have it working correctly in [this](https://github.com/nasa-gcn/gcn.nasa.gov/pull/272/commits/15cbefd151a9fc16a570337f0a9edcb07f0b3918) commit